lib/site-builder.js

```diff
@@ -56,8 +59,14 @@ SiteBuilder.prototype.build = function() {
 };
 
 function rsync(builder) {
-  return builder.commandRunner.run(config.rsync,
-    config.rsyncOpts.concat(['./', builder.configHandler.buildDestination]));
+  var generateRsyncOps = function(previousRsync, buildConfig) {
+    return previousRsync.then(function() {
+      return builder.commandRunner.run(config.rsync,
+        config.rsyncOpts.concat(['./', buildConfig.destination]));
+    });
+  };
```

Comment:
Ended up pulling out `generateRsyncOp` out, because the reduce function needs access to `builder`. I think the final result reads OK, though. Thoughts?